dotcom-rendering/src/web/browser/islands/doHydration.ts

```diff
@@ -24,16 +24,20 @@ export const doHydration = async (
 	const alreadyHydrated = element.dataset.guReady;
 	if (alreadyHydrated) return;
 
-	const { start, end } = initPerf(`hydrate-${name}`);
-	start();
+	const { start: startModule, end: endModule } = initPerf(`module-${name}`);
+	startModule();
 	await import(
 		/* webpackInclude: /\.importable\.tsx$/ */
 		/* webpackChunkName: "[request]" */
 		`../../components/${name}.importable`
 	)
 		.then((module) => {
+			const timeToLoad = endModule();
```

Comment:
I was thinking about this a little bit, and I'm not too sure, but is the duration from `startModule` to `endModule` the time taken to dynamically import the island AND the time taken for the promise to resolve? Could that be much longer than the time taken to load if there's a large queue of promises to resolve on the event loop?

Reply:
I’m not sure, but it’s fair to say that if it takes longer to load because of other parallel downloads, that should be taken into account.